spyderlib/plugins/shortcuts.py

```diff
@@ -187,186 +512,274 @@ def headerData(self, section, orientation, role=Qt.DisplayRole):
-                        or sh2.context == '_'):
+                   and (sh1.context == sh2.context or sh1.context == '_' or
+                        sh2.context == '_'):
                     conflicts.append((sh1, sh2))
         if conflicts:
             self.parent().show_this_page.emit()
             cstr = "\n".join(['%s <---> %s' % (sh1, sh2)
                               for sh1, sh2 in conflicts])
-            QMessageBox.warning(self, _( "Conflicts"),
+            QMessageBox.warning(self, _("Conflicts"),
                                 _("The following conflicts have been "
                                   "detected:")+"\n"+cstr, QMessageBox.Ok)
-        
+
     def save_shortcuts(self):
+        """Save shortcuts from table model."""
         self.check_shortcuts()
-        for shortcut in self.model.shortcuts:
+        for shortcut in self.source_model.shortcuts:
             shortcut.save()
-        
+
+    def show_editor(self):
+        """Create, setup and display the shortcut editor dialog."""
+        index = self.proxy_model.mapToSource(self.currentIndex())
+        row, column = index.row(), index.column()
+        shortcuts = self.source_model.shortcuts
+        context = shortcuts[row].context
+        name = shortcuts[row].name
+
+        sequence_index = self.source_model.index(row, SEQUENCE)
+        sequence = sequence_index.data()
+
+        dialog = ShortcutEditor(self, context, name, sequence, shortcuts)
+
+        if dialog.exec_():
+            new_sequence = dialog.new_sequence
+            self.source_model.setData(sequence_index, new_sequence)
+
+    def set_regex(self, regex=None, reset=False):
+        """Update the regex text for the shortcut finder."""
+        if reset:
+            text = ''
+        else:
+            text = self.finder.text().replace(' ', '').lower()
+
+        self.proxy_model.set_filter(text)
+        self.source_model.update_search_letters(text)
+
+        if self.last_regex != regex:
+            self.selectRow(0)
+        self.last_regex = regex
+
+    def next_row(self):
+        """Move to next row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.proxy_model.rowCount()
+        if row + 1 == rows:
+            row = -1
+        self.selectRow(row + 1)
+
+    def previous_row(self):
+        """Move to previous row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.proxy_model.rowCount()
+        if row == 0:
+            row = rows
+        self.selectRow(row - 1)
+
+    def keyPressEvent(self, event):
+        """Qt Override"""
+        key = event.key()
+        if key in [Qt.Key_Enter, Qt.Key_Return]:
+            self.show_editor()
+        elif key in [Qt.Key_Tab]:
+            self.parent().reset_btn.setFocus()
+        elif key in [Qt.Key_Backtab]:
+            self.finder.setFocus()
+        elif key in [Qt.Key_Up, Qt.Key_Down, Qt.Key_Left, Qt.Key_Right]:

```

Comment:
@blink1073 
I think we also need to adjust the code here as well from:

```python
        elif key in [Qt.Key_Tab]:
            self.parent().reset_btn.setFocus()
        elif key in [Qt.Key_Backtab]:
            self.finder.setFocus()
```

to

```python
        elif key in [Qt.Key_Tab]:
            self.finder.setFocus()
        elif key in [Qt.Key_Backtab]:
            self.parent().reset_btn.setFocus()
```